lib/Alchemy/Phrasea/WorkerManager/Worker/RecordEditWorker.php

```diff
@@ -30,122 +31,130 @@ public function __construct(WorkerRunningJobRepository $repoWorker, EventDispatc
+
+                    if (isset($rec['technicalsdatas']) && is_array($rec['technicalsdatas'])) {
+                        $record->insertOrUpdateTechnicalDatas($rec['technicalsdatas']);
+                    }
 
-                $record->write_metas();
+                    $newstat = $record->getStatus();
+                    $statbits = ltrim($statbits, 'x');
+                    if (!in_array($statbits, ['', 'null'])) {
+                        $mask_and = ltrim(str_replace(['x', '0', '1', 'z'], ['1', 'z', '0', '1'], $statbits), '0');
+                        if ($mask_and != '') {
+                            $newstat = \databox_status::operation_and_not($newstat, $mask_and);
+                        }
 
-                if ($statbits != '') {
-                    $this->getDataboxLogger($databox)
-                        ->log($record, \Session_Logger::EVENT_STATUS, '', '');
-                }
-                if ($editDirty) {
-                    $this->getDataboxLogger($databox)
-                        ->log($record, \Session_Logger::EVENT_EDIT, '', '');
+                        $mask_or = ltrim(str_replace('x', '0', $statbits), '0');
+
+                        if ($mask_or != '') {
+                            $newstat = \databox_status::operation_or($newstat, $mask_or);
+                        }
+
+                        $record->setStatus($newstat);
+                    }
+
+                    $record->write_metas();
+
+                    if ($statbits != '') {
+                        $this->getDataboxLogger($databox)
+                            ->log($record, \Session_Logger::EVENT_STATUS, '', '');
+                    }
+                    if ($editDirty) {
+                        $this->getDataboxLogger($databox)
+                            ->log($record, \Session_Logger::EVENT_EDIT, '', '');
+                    }
                 }
             }
-
-            // order to write metas for those records
-            $this->dispatcher->dispatch(WorkerEvents::RECORDS_WRITE_META,
-                new RecordsWriteMetaEvent(array_column($payload['mdsParams'], 'record_id'), $payload['databoxId'])
-            );
-
-            // tell that we have finished to work on edit
-            $this->repoWorker->reconnect();
-            $em->getConnection()->beginTransaction();
-            try {
-                $workerRunningJob->setStatus(WorkerRunningJob::FINISHED);
-                $workerRunningJob->setFinished(new \DateTime('now'));
-                $em->persist($workerRunningJob);
-                $em->flush();
-                $em->commit();
-            } catch (\Exception $e) {
-                $em->rollback();
+        } else {
+            $arg = json_decode($payload['data']);
+            $recordIds = array_column($arg->records, 'record_id');
+
+            // for now call record_adapter. no check, no acl, ...
+            foreach($arg->records as $rec) {
+                $r = $databox->get_record($rec->record_id);
```

Comment:
add try/catch, test r=null (like for mds)
the record could have been deleted before worker.